inc/functions.php

```diff
@@ -8239,3 +8239,22 @@ function copy_file_to_cdn($file_path = '', &$uploaded_path = null)
 
 	return $success;
 }
+
+/**
+ * Validate an url
+ *
+ * @param string $url The url to validate.
+ * @param bool $relative_path Whether or not the url could be a relative path.
+ *
+ * @return bool Whether this is a valid url.
+ */
+function my_validate_url($url, $relative_path=false)
+{
+	if($relative_path && my_substr($url, 0, 1) == '/' || preg_match('^(?:(?:https?|ftp)://)(?:\S+(?::\S*)?@|\d{1,3}(?:\.\d{1,3}){3}|(?:(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)(?:\.(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)*(?:\.[a-z\x{00a1}-\x{ffff}]{2,6}))(?::\d+)?(?:[^\s]*)?$', $url))
```

Comment:
Yep, you are correct. I updated the code, tested and seems to work now except for two false positives.

According to the above page the following should be invalid URLs:
```
http://a.b--c.de/
http://www.foo.bar./
```

But even the code supplied by them marks them as valid. Tough the first one seems valid to myself so..